### ISSUE TITLE
Tooltips: "data[config] is not a function" error

### ISSUE THREAD

**Issue by denizozger** [twbs/bootstrap#17544]
Running ```$(this).tooltip('destroy')``` inside a jQuery function throws this ```Uncaught TypeError: data[config] is not a function``` error.

```
$(function () {
  $('[data-toggle="tooltip"]').tooltip();
  
  $('[data-toggle="tooltip"]').each(function() {
    $(this).tooltip('destroy');
  });
})
```

Demonstration is below: http://codepen.io/denizozger/pen/ojXKNg

Here is the same code with Bootstrap 3.3.5: http://codepen.io/denizozger/pen/RWPXwR

**Comment by Johann-S** [comment]
I think it's just a bug in the documentation
because to destroy a tooltip you have to use ``$(this).tooltip('dispose')`` and not destroy
as it wrote in : http://v4-alpha.getbootstrap.com/components/tooltips/#tooltipdestroy

**Comment by denizozger** [comment]
I can confirm ```$(this).tooltip('dispose')``` works, however in current documentation ```destroy``` is the method to use (http://getbootstrap.com/javascript/#tooltips), so ```destroy``` needs to remain for backward compatibility purposes.

**Comment by Johann-S** [comment]
I'm not sure if we have to keep ``destroy`` in V4.
New versions comes with changes.

**Comment by Johann-S** [comment]
Any advice about that @cvrebert ?

**Comment by cvrebert** [comment]
Yeah, looks like the v4 docs weren't updated when the method was renamed. They need to be corrected.
Backward compatibility isn't a concern in v4, so there we won't be adding any `destroy` method.

**Comment by cvrebert** [comment]
Fixed by #17613.
Also filed #17612 about showing a better error message in this case.
